### ISSUE TITLE
VideoDecoderTest.FlushAfterInitialize fails

### ISSUE THREAD

**Issue by EwanRoycroft** [bbc/chromium#13]
### Description
`FlushAfterInitialize` fails when testing the `VideoDecoder` pipeline with `video_decode_accelerator_tests`.

### More information
| Attribute | Value |
| --- | --- |
| **Version** | 84+ |
| **Build** | Release |
| **Platform** | Ozone-GBM |
| **Target** | `video_decode_accelerator_tests` |
| **Device** | Raspberry Pi 4B (4GB) |
| **OS** | Raspbian 10 "buster" |
| **Kernel** | 5.4.51-v7l+ |

### Steps to reproduce
Run `video_decode_accelerator_tests` for the `VideoDecoder` pipeline.

#### GN arguments
```
use_ozone = true
ozone_platform_x11 = false
ozone_platform_gbm = true
ozone_platform_headless = false
use_vc4_minigbm = true

ffmpeg_branding = "Chrome"
proprietary_codecs = true
use_v4l2_codec = true
use_v4l2_non_standard = false
use_v4l2_stateless_decoder = false

target_os = "chromeos"
target_cpu = "arm"
use_sysroot = true
target_sysroot = "//build/linux/raspbian_arm-sysroot"

google_api_key = "no"
google_default_client_id = "no"
google_default_client_secret = "no"

is_debug = false
is_component_build = false
enable_nacl = false
symbol_level = 0
blink_symbol_level = 0
```

#### Command
```shell
./video_decode_accelerator_tests ../../media/test/data/test-25fps.h264 --use_vd [--vmodule=../../media/*=4]
```

### Terminal output
```
[1102/174446.759711:ERROR:video_player.cc(205)] Timeout while waiting for 'FlushDone' event
../../media/gpu/video_decode_accelerator_tests.cc:149: Failure
Value of: tvp->WaitForFlushDone()
  Actual: false
Expected: true
Stack trace:
#0 0x000000609840 media::test::VideoDecoderTest_FlushAfterInitialize_Test::TestBody()

../../media/gpu/test/video_player/video_decoder_client.cc:279: Failure
Expected equality of these values:
  decoder_client_state_
    Which is: 4-byte object <03-00 00-00>
  VideoDecoderClientState::kIdle
    Which is: 4-byte object <01-00 00-00>
Stack trace:
#0 0x00000088d9ac media::test::VideoDecoderClient::PlayTask()
#1 0x0000008aa70e base::TaskAnnotator::RunTask()
#2 0x0000008b3c8c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#3 0x0000008b3a66 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#4 0x0000008b3f56 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#5 0x0000008c238a base::MessagePumpDefault::Run()
#6 0x0000008b4104 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#7 0x0000008a6752 base::RunLoop::Run()
#8 0x0000008c1fc4 base::Thread::ThreadMain()

[1102/174517.133984:ERROR:video_player.cc(205)] Timeout while waiting for 'FlushDone' event
../../media/gpu/video_decode_accelerator_tests.cc:151: Failure
Value of: tvp->WaitForFlushDone()
  Actual: false
Expected: true
Stack trace:
#0 0x0000006098e4 media::test::VideoDecoderTest_FlushAfterInitialize_Test::TestBody()

../../media/gpu/video_decode_accelerator_tests.cc:153: Failure
Expected equality of these values:
  tvp->GetFlushDoneCount()
    Which is: 0
  2u
    Which is: 2
Stack trace:
#0 0x000000609984 media::test::VideoDecoderTest_FlushAfterInitialize_Test::TestBody()

../../media/gpu/video_decode_accelerator_tests.cc:154: Failure
Expected equality of these values:
  tvp->GetFrameDecodedCount()
    Which is: 0
  g_env->Video()->NumFrames()
    Which is: 250
Stack trace:
#0 0x000000609a1e media::test::VideoDecoderTest_FlushAfterInitialize_Test::TestBody()
```

**Comment by EwanRoycroft** [comment]
This is due to #11. Temporarily fixed by [b67e81e](https://github.com/bbc/chromium/commit/b67e81e51c5c68f0fd274c2eca277d0ab83d0e61).

**Comment by EwanRoycroft** [comment]
[b67e81e](https://github.com/bbc/chromium/commit/b67e81e51c5c68f0fd274c2eca277d0ab83d0e61) appears to break `VideoDecoderTest.FlushAtEndOfStream`:
```
../../media/gpu/video_decode_accelerator_tests.cc:146: Failure
Expected equality of these values:
  tvp->GetFrameDecodedCount()
    Which is: 175
  g_env->Video()->NumFrames()
    Which is: 250
Stack trace:
#0 0x0000005eb6be media::test::VideoDecoderTest_FlushAtEndOfStream_Test::TestBody()
```
It also fails to fix `FlushAfterInitialize` if another test has run first. Presumably `has_initialised` has not been reset to `false` for the new test so the flush never completes.

**Comment by EwanRoycroft** [comment]
Appears to have been fixed on latest `px-rpi` branch using kernel v5.9.

**Comment by EwanRoycroft** [comment]
Test passes in Chromium 89.0.4389.72+.
